src/autocomplete/shared/extract-statement-positions-from-query.ts

```diff
@@ -0,0 +1,67 @@
+import {LexerConstructor} from './autocomplete-types';
+import {Token, tokenize} from './tokenize';
+import type {Lexer as LexerType} from 'antlr4ng';
+
+export interface StatementPosition {
+    startIndex: number;
+    endIndex: number;
+}
+
+export function extractStatementPositionsFromQuery<L extends LexerType>(
+    query: string,
+    Lexer: LexerConstructor<L>,
+    symbolicNames: (string | null)[],
+    whiteSpaceToken: number,
+    emptyTokens: number[],
+    closeStatementToken: number,
```

Comment:
I'd rename these to
`emptySpaceTokens` or `separatorTokens`, because `emptyTokens` sounds like tokens that are undefined
`endStatementToken`, because `close` doesn't fit here